### PR TITLE
Add parser changes for isolated objects

### LINKED ISSUE #25569: Parser changes for isolated objects
**Description:**
Need to allow `isolated` with class definitions.

```ballerina
isolated class Foo {

}
```

Need to also confirm if `isolated` needs to be allowed with object constructor expressions and support the same if so.

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose

- Add isolated qualifier support for module-class-defn and object-type-descriptor

Fixes #25569

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -1022,7 +1022,8 @@ public BLangNode transform(ObjectFieldNode objFieldNode) {
         Optional<Node> doc = getDocumentationString(objFieldNode.metadata());
         simpleVar.markdownDocumentationAttachment = createMarkdownDocumentationAttachment(doc);
 
-        addRedonlyQualifier(objFieldNode.readonlyKeyword(), objFieldNode.typeName(), simpleVar);
+        // TODO: add changes for `final`
+//        addRedonlyQualifier(objFieldNode.readonlyKeyword(), objFieldNode.typeName(), simpleVar);
```
Comment (SupunS) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1026]: Lets remove this, if its no longer needed.

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1026]: @SupunS PR is not yet ready to be reviewed. Waiting for PR #25784 to be merged to clean up old commits from that PR. Also have to do small refactoring. Will let you knw once ready. :)

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/latest/ClassDefContextTest.java [hunk 2]
```diff
@@ -41,6 +41,6 @@ public String getTestResourceDir() {
 
     @Override
     public List<String> skipList() {
-        return Arrays.asList("config3.json", "config8.json");
+        return Arrays.asList("config3.json", "config8.json", "config12.json", "config13.json");
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/latest/ClassDefContextTest.java:44]: Can we create an issue to track this test disabling?

Comment (lochana-chathura) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/latest/ClassDefContextTest.java:44]: Issue created https://github.com/ballerina-platform/ballerina-lang/issues/26017

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 3]
```diff
@@ -4625,56 +4662,150 @@ private STNode parseObjectConstructorExpression(STNode annots) {
...
+                return STNodeFactory.createEmptyNodeList();
+            default:
+                recover(nextToken, ParserRuleContext.OBJECT_TYPE_QUALIFIER);
+                return parseObjectTypeQualifiers();
+        }
+
+        return parseObjectTypeNextQualifiers(firstQualifier);
+    }
+
+    private STNode parseObjectTypeNextQualifiers(STNode firstQualifier) {
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:4770]: Similar to the prev one, I think we don't need the `parseObjectTypeQualifiers()` method

File: compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java [hunk 4]
```diff
@@ -4625,56 +4662,150 @@ private STNode parseObjectConstructorExpression(STNode annots) {
...
+                return STNodeFactory.createEmptyNodeList();
+            default:
+                recover(nextToken, ParserRuleContext.OBJECT_CONSTRUCTOR_QUALIFIER);
+                return parseObjectConstructorQualifiers();
+        }
+
+        return parseObjectConstructorNextQualifiers(firstQualifier);
+    }
+
+    private STNode parseObjectConstructorNextQualifiers(STNode firstQualifier) {
```
Comment (SupunS) [inline compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java:4697]: Do we need two methods?
I think the one with the loop would be sufficient to handle all cases.

**Review by SupunS — CHANGES_REQUESTED** [review]

**Review by SupunS — APPROVED** [review]

**Pull request merged**